### ISSUE TITLE
Flock does not propagate signals

### ISSUE THREAD

**Issue by fxposter** [kubernetes-incubator/bootkube#502]
This is probably related to #495, but I'd like to raise a question regarding the `flock` itself.
If `flock` gets any signal (for example, `SIGTERM`) and it does already started the command that is passed to it - `flock` will die itself, but does not send the signals to the running process.
This actually means that `apiserver` and other components that are run through `flock` will be actually `SIGKILL`-ed when someone will try stopping the `apiserver`.
Were there any thoughts about this?

**Comment by fxposter** [comment]
Since v2.29 `flock` supports `--no-fork` option (https://github.com/karelzak/util-linux/blob/stable/v2.29/sys-utils/flock.c#L70), which will probably solve the issue, but it is not in the apiserver images, which use debian jessie, which have v2.25.

**Comment by rphillips** [comment]
Debian 9 (stretch) does contain the fixed flock functionality (util-linux 2.29.2). We could upgrade the BASEIMAGE from jessie to stretch, or perhaps use a sidecar container containing the upgraded version of util-linux. Thoughts?

**Comment by diegs** [comment]
We have been trying to eliminate the flock for the apiserver, see #616

Might be easier to finish that rather than changing the upstream debian image and waiting for that to make it to a current release.

**Comment by redbaron** [comment]
There is Python however, so flock can be implemented using following oneliner:

```
python -c 'import fcntl,os,sys; f=open(sys.argv[1],"w"); fcntl.lockf(f, fcntl.LOCK_EX); os.execlp(sys.argv[2],sys.argv[2],*sys.argv[sys.argv.index("--")+1:])' /var/lock/api-server.lock /hyperkube -- apiserver ....
```

**Comment by redbaron** [comment]
Just tested with following manifests and it works well:

```
      containers:
      - name: kube-apiserver
        image: gcr.io/google_containers/hyperkube:v1.8.3
        command:
        - python
        - -c
        - 'import fcntl,os,sys; f=open(sys.argv[1],"w"); fcntl.lockf(f, fcntl.LOCK_EX); os.execlp(sys.argv[2],sys.argv[2],*sys.argv[sys.argv.index("--")+1:])'
        - /var/lock/api-server.lock
        - /hyperkube
        - --
        - apiserver
```

Running apiserver process is only process in container and it holds lock:
```
hyperkube 3246 root    3wW     REG   0,21         0   27753 /run/lock/api-server.lock
```

**Comment by diegs** [comment]
I believe @rphillips has done work to migrate the hyperkube images to debian jessie. I am retrying my PR to remove the flock, but otherwise we can probably add `--no-fork` once the jessie hyperkubes are released.

**Comment by rphillips** [comment]
*stretch hyperkubes will be in k8s 1.9, where we can add the ```--no-fork```.

**Comment by diegs** [comment]
Fixed by #616
